graph-commons/src/main/scala/io/renku/data/Message.scala

```diff
@@ -0,0 +1,212 @@
+        blankToNone(message)
+          .map(toSingleLine)
+          .getOrElse(throw new IllegalArgumentException("Message cannot be instantiated with a blank String"))
+      }
+  }
+
+  object Error {
+
+    def fromJsonUnsafe(errorMessage: Json): Message =
+      if (errorMessage.isNull || errorMessage == Json.obj())
+        throw new IllegalArgumentException("Message cannot be an empty Json")
+      else
+        JsonMessage(errorMessage, Severity.Error)
+
+    def apply(value: String Refined NonEmpty): Message =
+      StringMessage(value.value, Severity.Error)
+
+    def unsafeApply(errorMessage: String): Message =
+      toStringMessageUnsafe(Severity.Error) {
+        blankToNone(errorMessage)
+          .map(toSingleLine)
+          .getOrElse(throw new IllegalArgumentException("Message cannot be instantiated with a blank String"))
+      }
+
+    def fromExceptionMessage(exception: Throwable): Message =
+      toStringMessageUnsafe(Severity.Error) {
+        blankToNone(exception.getMessage)
+          .fold(ifEmpty = exception.getClass.getName)(toSingleLine)
+      }
+
+    def fromStackTrace(exception: Throwable): Message =
+      toStringMessageUnsafe(Severity.Error) {
+        blankToNone {
+          val sw = new StringWriter
+          exception.printStackTrace(new PrintWriter(sw))
+          sw.toString
+        }.fold(ifEmpty = exception.getClass.getName)(toSingleLine)
+      }
+
+    def fromMessageAndStackTraceUnsafe(message: String, exception: Throwable): Message =
+      toStringMessageUnsafe(Severity.Error) {
+        Option(exception)
+          .flatMap { e =>
+            blankToNone {
+              val sw = new StringWriter
+              e.printStackTrace(new PrintWriter(sw))
+              s"$message; $sw"
+            }
+          }
+          .fold(ifEmpty = message)(toSingleLine)
+      }
+  }
+
+  private def blankToNone(message: String): Option[String] =
+    Option(message).map(_.trim) >>= {
+      case ""       => None
+      case nonBlank => Some(nonBlank)
+    }
+
+  private lazy val toSingleLine: String => String =
+    _.split('\n').map(_.trim).mkString("", "; ", "")
+
+  private def toStringMessageUnsafe(severity: Severity): String => Message.StringMessage =
+    _.trim match {
+      case ""       => throw new IllegalArgumentException(show"$severity message cannot be blank")
+      case nonBlank => StringMessage(nonBlank, severity)
+    }
+
+  private final case class StringMessage(value: String, severity: Severity) extends Message {
+    override type T = String
+    override lazy val show: String = value
+  }
+  private final case class JsonMessage(value: Json, severity: Severity) extends Message {
+    override type T = Json
+    override lazy val show: String = value.noSpaces
+  }
+
+  implicit def show[T <: Message]: Show[T] = Show.show[T] {
+    case m: Message.StringMessage => m.show
+    case m: Message.JsonMessage   => m.show
+  }
+
+  object Codecs {
```

Comment:
Why not putting these into the companion objects? Then there wouldn't be a need to add some other import?

Reply:
Done